test_conformance/device_execution/enqueue_profiling.cpp

```diff
@@ -0,0 +1,169 @@
+        CL_QUEUE_SIZE, maxQueueSize, 0
+    };
+
+    dev_queue = clCreateCommandQueueWithProperties(
+        context, device, dev_queue_prop_def, &err_ret);
+    test_error(err_ret,
+               "clCreateCommandQueueWithProperties(CL_QUEUE_DEVICE|CL_QUEUE_"
+               "DEFAULT) failed");
+
+    cl_queue_properties host_queue_prop_def[] = { CL_QUEUE_PROPERTIES,
+                                                  CL_QUEUE_PROFILING_ENABLE,
+                                                  0 };
+
+    host_queue = clCreateCommandQueueWithProperties(
+        context, device, host_queue_prop_def, &err_ret);
+    test_error(err_ret,
+               "clCreateCommandQueueWithProperties(CL_QUEUE_DEVICE|CL_QUEUE_"
+               "DEFAULT) failed");
+
+    cl_int status;
+    size_t size = 1;
+    cl_int result = 0;
+
+    clMemWrapper res_mem;
+    clProgramWrapper program;
+    clKernelWrapper kernel;
+
+    cl_event kernel_event;
+
+    err_ret = create_single_kernel_helper_with_build_options(
+        context, &program, &kernel, 1, &enqueue_multi_level,
+        "enqueue_multi_level", "-cl-std=CL2.0");
+    if (check_error(err_ret, "Create single kernel failed")) return -1;
+
+    res_mem = clCreateBuffer(context, CL_MEM_READ_WRITE | CL_MEM_COPY_HOST_PTR,
+                             sizeof(result), &result, &err_ret);
+    test_error(err_ret, "clCreateBuffer() failed");
+
+    err_ret = clSetKernelArg(kernel, 0, sizeof(res_mem), &res_mem);
+    test_error(err_ret, "clSetKernelArg(0) failed");
+
+    for (int level = 0; level < max_nestingLevel; level++)
+    {
+        err_ret = clSetKernelArg(kernel, 1, sizeof(level), &level);
+        test_error(err_ret, "clSetKernelArg(1) failed");
+
+        err_ret = clEnqueueNDRangeKernel(host_queue, kernel, 1, NULL, &size,
+                                         &size, 0, NULL, &kernel_event);
+        test_error(err_ret,
+                   "clEnqueueNDRangeKernel('enqueue_multi_level') failed");
+
+        err_ret = clEnqueueReadBuffer(host_queue, res_mem, CL_TRUE, 0,
+                                      sizeof(result), &result, 0, NULL, NULL);
+        test_error(err_ret, "clEnqueueReadBuffer() failed");
+
+        err_ret =
+            clGetEventInfo(kernel_event, CL_EVENT_COMMAND_EXECUTION_STATUS,
+                           sizeof(status), &status, NULL);
+        test_error(err_ret, "clGetEventInfo() failed");
+
+        cl_ulong end;
+        err_ret = clGetEventProfilingInfo(
+            kernel_event, CL_PROFILING_COMMAND_END, sizeof(end), &end, NULL);
+        test_error(err_ret, "clGetEventProfilingInfo() failed");
+
+        cl_ulong complete;
+        err_ret =
+            clGetEventProfilingInfo(kernel_event, CL_PROFILING_COMMAND_COMPLETE,
+                                    sizeof(complete), &complete, NULL);
+        test_error(err_ret, "clGetEventProfilingInfo() failed");
+
+        if (check_error(status, "Kernel execution status %d", status))
+            return status;
```

Comment:
Might be better to do that before the calls to `clGetEventProfilingInfo` since these would fail if the status was not `CL_COMPLETE`.